pkg/server/grpc/tracee.go

```diff
@@ -83,3 +118,189 @@ func convertDefinitionToProto(d events.Definition) *pb.EventDefinition {
 		Tags:        d.GetSets(),
 	}
 }
+
+func convertTraceeEventToProto(e trace.Event) *pb.Event {
+	process := getProcess(e)
+	container := getContainer(e)
+	k8s := getK8s(e)
+
+	var eventContext *pb.Context
+	if process != nil || container != nil || k8s != nil {
+		eventContext = &pb.Context{
+			Process:   process,
+			Container: container,
+			K8S:       k8s,
+		}
+	}
+
+	event := &pb.Event{
+		Id:   uint32(e.EventID),
+		Name: e.EventName,
+		Policies: &pb.Policies{
+			Matched: e.MatchedPolicies,
+		},
+		Context: eventContext,
+		Threat:  getThreat(e),
+	}
+
+	if e.Timestamp != 0 {
+		event.Timestamp = timestamppb.New(time.Unix(int64(e.Timestamp), 0))
+	}
+
+	return event
+}
+
+func getProcess(e trace.Event) *pb.Process {
+	var userStackTrace *pb.UserStackTrace
+
+	if len(e.StackAddresses) > 0 {
+		userStackTrace = &pb.UserStackTrace{
+			Addresses: e.StackAddresses,
+		}
+	}
+
+	var thredStartTime *timestamp.Timestamp
+	if e.ThreadStartTime != 0 {
+		thredStartTime = timestamppb.New(time.Unix(int64(e.ThreadStartTime), 0))
+	}
+
+	return &pb.Process{
+		// Executable
+		EntityId:      wrapperspb.UInt32(e.EntityID),
+		Pid:           wrapperspb.UInt32(uint32(e.HostProcessID)),
+		NamespacedPid: wrapperspb.UInt32(uint32(e.ProcessID)),
+		RealUser: &pb.User{
+			Id: wrapperspb.UInt32(uint32(e.UserID)),
+		},
+		Thread: &pb.Thread{
+			Start:          thredStartTime,
+			Name:           e.ProcessName,
+			Tid:            wrapperspb.UInt32(uint32(e.HostThreadID)),
+			NamespacedTid:  wrapperspb.UInt32(uint32(e.ThreadID)),
+			Syscall:        e.Syscall,
+			Compat:         e.ContextFlags.ContainerStarted,
+			UserStackTrace: userStackTrace,
+		},
+		Parent: &pb.Parent{
+			Pid:           wrapperspb.UInt32(uint32(e.HostParentProcessID)),
+			NamespacedPid: wrapperspb.UInt32(uint32(e.ParentProcessID)),
+		},
+	}
+}
+
+func getContainer(e trace.Event) *pb.Container {
+	if e.ContainerID == "" && e.Container.Name == "" {
+		return nil
+	}
+
+	container := &pb.Container{
+		Id:   e.Container.ID,
+		Name: e.Container.Name,
+	}
```

Comment:
nit: first you use `e.ContainerID` and then `e.Container.ID` - better stick with one of those